presto-common/src/main/java/com/facebook/presto/common/block/AbstractMapBlock.java

```diff
@@ -500,26 +499,6 @@ public void loadHashTables(int positionCount, int[] offsets, boolean[] mapIsNull
             }
             set(hashTables);
         }
-
-        @Override
-        public boolean equals(Object obj)
-        {
-            if (this == obj) {
-                return true;
-            }
-            if (obj == null || getClass() != obj.getClass()) {
-                return false;
-            }
-            HashTables other = (HashTables) obj;
-            return Arrays.equals(this.hashTables, other.hashTables) &&
-                    this.expectedHashTableCount == other.expectedHashTableCount;
-        }
-
-        @Override
-        public int hashCode()
-        {
-            return Objects.hash(Arrays.hashCode(hashTables), expectedHashTableCount);
-        }
```

Comment:
Same, can we add comment to this class with javadoc or whatever to explain the rationale?